std/traits.d

```diff
@@ -6839,3 +6839,348 @@ unittest
     struct BoolWrapper { bool value; }
     static assert(!ifTestable!(bool, a => BoolWrapper(a)));
 }
+
+/**
+ * Detect whether $(D X) is a type or not.
+ *
+ * Returns:
+ *     `bool`
```

Comment:
isEnum, isStruct, isUnion, etc. are very useful in combination with Filter and the like from std.meta. For example:
```D
alias allEnuns = Filter!(isEnum, __traits(allMembers, mixin(__MODULE__)));

size_t sizeOfStructs(T...)()
    if (allSatisfy!(isStruct, T))
{ /* ...*/ }
```